src/annotator/index.js

```diff
@@ -41,8 +46,11 @@ function init() {
 
   const hostFrame = annotatorConfig.subFrameIdentifier ? window.parent : window;
 
+  /** @type {Sidebar|undefined} */
   let sidebar;
```

Comment:
I expected that TypeScript would infer the type of these variables automatically based on subsequent assignments. It appears however that it doesn't when the variable is accessed in a callback.

In this code:

```js
sidebar = new Sidebar(document.body, eventBus, sidebarConfig);
notebook = new Notebook(document.body, eventBus, getConfig('notebook'));

portProvider.on('frameConnected', (source, port) =>
  /** @type {Sidebar} */ (sidebar).onFrameConnected(source, port)
);
```

The type of `sidebar` will be inferred as `Sidebar` in the top-level function after the assignment, but it is inferred as `any` within the callback.

We could actually avoid the need for `let` here. The only use of `sidebar` outside the `if (hostFrame === window) { ... }` block is to call `destroy` later. The same applies to the other conditionally-defined variables. We could collect all of these in a `Destroyable[]` and just loop over the array to cleanup. This would allow the variables to be scoped to the block where they are assigned and assigned to a `const` binding.